project/AkkaBuild.scala

```diff
@@ -21,14 +24,25 @@ object AkkaBuild {
 
   lazy val buildSettings = Dependencies.Versions ++ Seq(
     organization := "com.typesafe.akka",
-    // use the same value as in the build scope, so it can be overriden by stampVersion
+    // use the same value as in the build scope
     version := (version in ThisBuild).value)
 
+  lazy val currentDateTime = DateTimeFormatter
+    .ofPattern("yyyyMMdd-HHmmss")
+    .format(ZonedDateTime.now(ZoneOffset.UTC))
```

Comment:
Taking from my comment in https://github.com/akka/akka/issues/26958#issuecomment-493454817, I think a less fragile implementation would be:

```scala
  val timestamp = sys.props.getOrElseUpdate(
    "akka.build.timestamp",
    java.time.format.DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss").format(
      java.time.ZonedDateTime.now(java.time.ZoneOffset.UTC)
    )
  )
```

It also benefits from allowing it to be defined externally.

Reply:
added in 30c0b5e